src/Zcu/PerThread.zig

```diff
@@ -1950,6 +1950,70 @@ pub fn importPkg(pt: Zcu.PerThread, mod: *Module) Allocator.Error!Zcu.ImportFile
     };
 }
 
+fn eqlIgnoreCase(a: []const u8, b: []const u8) bool {
+    if (builtin.os.tag == .windows)
+        return std.os.windows.eqlIgnoreCaseWtf8(a, b);
+
+    var a_utf8_it = std.unicode.Utf8View.initUnchecked(a).iterator();
+    var b_utf8_it = std.unicode.Utf8View.initUnchecked(b).iterator();
+    while (true) {
+        const a_cp = a_utf8_it.nextCodepoint() orelse break;
+        const b_cp = b_utf8_it.nextCodepoint() orelse return false;
+        if (a_cp != b_cp) {
+            const a_upper = std.ascii.toUpper(std.math.cast(u8, a_cp) orelse return false);
+            const b_upper = std.ascii.toUpper(std.math.cast(u8, b_cp) orelse return false);
+            if (a_upper != b_upper)
+                return false;
+        }
+    }
+    if (b_utf8_it.nextCodepoint() != null) return false;
```

Comment:
Note that this has no practical difference to `std.ascii.eqlIgnoreCase`, since UTF-8 multibyte codepoints will never have bytes within the ASCII range.

(but see https://github.com/ziglang/zig/pull/23163#discussion_r1990057336)